### PR TITLE
[Platform API][SFP] Compare applicable data against values from platform.json

### PR DESCRIPTION
**Pull request by jleveque**
Summary:

For SFP tests where known facts are available in the platform.json file, compare actual values against those facts

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?

To improve the quality of the SFP platform API test

#### How did you do it?

For SFP tests where known facts are available in the platform.json file, compare actual values against those facts 

#### How did you verify/test it?

Run SFP tests on a device with a populated platform.json present

#### Any platform specific information?

Platform must have a populated platform.json present

### CODE REVIEW

**Review by yxieca — APPROVED** [review]
Maybe have a future change to separate the fact comparison test to a new test case so that they can be skipped instead of generating warning?

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
@yxieca: We could separate the fact comparison test to a new test case, but the idea is is to raise attention to the fact that the vendor has not yet provided a platform.json file for the platform. I worry that skipping tests might make it go unnoticed.

**Comment by yxieca** [comment]
> @yxieca: We could separate the fact comparison test to a new test case, but the idea is is to raise attention to the fact that the vendor has not yet provided a platform.json file for the platform. I worry that skipping tests might make it go unnoticed.

good point. in this case, then, separate test case and fail the test case when platform.json is missing is more explicit than warning.

**Comment by jleveque** [comment]
> in this case, then, separate test case and fail the test case when platform.json is missing is more explicit than warning.

I was planning to eventually make the platform.json check mandatory, once vendors had time to implement, but I wasn't planning to separate the test cases, as it will make the test run much longer. Maybe we should just make both the platform.json and inventory file checks mandatory now?

**Comment by jleveque** [comment]
Retest this please

File: tests/platform_tests/api/test_sfp.py [hunk 1]
```diff
@@ -106,6 +109,34 @@ def setup(self, platform_api_conn):
...
+
+    #
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value, sfp_idx):
+        expected_value = None
+
+        if self.chassis_facts:
+            expeceted_sfps = self.chassis_facts.get("sfps")
```
Comment (vdahiya12) [inline tests/platform_tests/api/test_sfp.py:121]: expected_sfps typo, 
Spelling mistake please update

**Review by vdahiya12 — APPROVED** [review]

**Comment by jleveque** [comment]
Retest this please

**Pull request merged**